picard/ui/metadatabox.py

```diff
@@ -317,11 +323,18 @@ def contextMenuEvent(self, event):
         menu.exec_(event.globalPos())
         event.accept()
 
+    def add_to_preserved_tags(self, name):
+        preserved_tags = [tag for tag in
+                          map(lambda x: x.strip(), config.setting['preserved_tags'].split(','))
+                          if tag != ""]
+        preserved_tags.append(name)
+        config.setting['preserved_tags'] = ", ".join(set(preserved_tags))
```

Comment:
Using a `set` here doesn't preserve the order of elements in the setting. It's desirable to keep that because not all tags are conceptually independent, so you might have `replaygain_album_gain`, `replaygain_track_gain` that you enter one after another, but building the set shuffles the order. Fortunately, there's a `uniquify` function in `picard.util` that removes duplicates and keeps the order of elements.